sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs

```diff
@@ -138,5 +149,16 @@ await _sessionReceiver.AbandonMessageAsync(message, propertiesToModify, cancella
             .ConfigureAwait(false);
             message.IsSettled = true;
         }
+
+        /// <summary>
+        /// Marks the session that is being processed as closed. No new receives will be initiated for the session before the
+        /// session is closed. Any already received messages will still be delivered to the user message handler, and in-flight message handlers
+        /// will be allowed to complete. Messages will still be completed automatically if <see cref="ServiceBusSessionProcessorOptions.AutoCompleteMessages"/>
+        /// is <value>true</value>.
+        /// The session may end up being reopened for processing immediately after closing if there are messages remaining in the session (
+        /// This depends on what other session messages may be in the queue or subscription).
+        /// </summary>
+        public virtual void MarkSessionClosed() =>
```

Comment:
Should we consider adding a boolean flag to the args for something like `SessionMarkedAsClosed` or `SessionCloseRequested` or similar?

Reply:
That seems reasonable.

Reply:
Actually, the existing CancellationToken property can be used for the same purpose. It will be canceled once the user calls MarkSessionClosed.